var/spack/repos/builtin/packages/visit/package.py

```diff
@@ -95,6 +95,7 @@ class Visit(CMakePackage):
     patch("cmake-findvtkh-3.3.patch", when="@3.3.0:3.3.2+vtkm")
     patch("cmake-findjpeg.patch", when="@3.1.0:3.2.2")
     patch("cmake-findjpeg-3.3.patch", when="@3.3.0")
+    patch("0001-fix-missing-header-includes-for-QSurfaceFormat.patch", when="@3.3.3+gui")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
I added a comment.

I don't know why it's needed, since it seems it would fail to compile without this in any circumstances where `+gui` is enabled. It wasn't taken from anywhere -- I had to figure out this header was missing from the `incomplete type` error messages produced by the compiler.

Reply:
Even just the previous comment is fine. I usually ask for comments above patches, so that people reading a package.py file can get a sense of what the patch is for without downloading it / opening it too.